ui/app/serializers/connector.js

```diff
@@ -20,11 +20,22 @@ export default class ConnectorSerializer extends ApplicationSerializer {
       type: Object.keys(CONNECTOR_TYPE_MAP).find(
         (key) => CONNECTOR_TYPE_MAP[key] === snapshot.record.type
       ),
-      plugin: snapshot.record.plugin,
+      plugin: snapshot.record.plugin.get('id'),
       pipeline_id: snapshot.record.pipeline.get('id'),
     };
   }
 
+  extractRelationship(modelName, value) {
+    if (modelName === 'plugin' && value) {
+      return {
+        id: value,
+        type: modelName,
+      };
+    }
+
+    super.extractRelationship(modelName, value);
```

Comment:
Would this also require a `return` statement or is this irrelevant in this case?

```js
return super.extractRelationship(modelName, value);
```